Libraries/LibWeb/IndexedDB/Internal/Algorithms.cpp

```diff
@@ -864,4 +866,118 @@ ErrorOr<GC::Ref<Key>> convert_a_value_to_a_multi_entry_key(JS::Realm& realm, JS:
+        // If Type(value) is String, and identifier is "length"
+        if (value.is_string() && identifier == "length") {
+            // Let value be a Number equal to the number of elements in value.
+            value = JS::Value(value.as_string().utf16_string_view().length_in_code_units());
+        }
+
+        // If value is an Array and identifier is "length"
+        else if (value.is_object() && is<JS::Array>(value.as_object()) && identifier == "length") {
+            // Let value be ! ToLength(! Get(value, "length")).
+            value = JS::Value(MUST(length_of_array_like(realm.vm(), value.as_object())));
+        }
+
+        // If value is a Blob and identifier is "size"
+        else if (value.is_object() && is<FileAPI::Blob>(value.as_object()) && identifier == "size") {
+            // Let value be value’s size.
+            value = JS::Value(static_cast<FileAPI::Blob&>(value.as_object()).size());
+        }
+
+        // If value is a Blob and identifier is "type"
+        else if (value.is_object() && is<FileAPI::Blob>(value.as_object()) && identifier == "type") {
+            // Let value be a String equal to value’s type.
+            value = JS::PrimitiveString::create(realm.vm(), static_cast<FileAPI::Blob&>(value.as_object()).type());
+        }
+
+        // If value is a File and identifier is "name"
+        else if (value.is_object() && is<FileAPI::File>(value.as_object()) && identifier == "name") {
+            // Let value be a String equal to value’s name.
+            value = JS::PrimitiveString::create(realm.vm(), static_cast<FileAPI::File&>(value.as_object()).name());
+        }
+
+        // If value is a File and identifier is "lastModified"
+        else if (value.is_object() && is<FileAPI::File>(value.as_object()) && identifier == "lastModified") {
+            // Let value be a Number equal to value’s lastModified.
+            value = JS::Value(static_cast<double>(static_cast<FileAPI::File&>(value.as_object()).last_modified()));
+        }
+
+        // Otherwise
+        else {
+            // 1. If Type(value) is not Object, return failure.
+            if (!value.is_object())
+                return Error::from_string_literal("Value is not an object");
+
+            // 2. Let hop be ! HasOwnProperty(value, identifier).
+            auto hop = MUST(value.as_object().has_own_property(identifier));
+
+            // 3. If hop is false, return failure.
+            if (!hop)
+                return Error::from_string_literal("Property does not exist");
+
+            // 4. Let value be ! Get(value, identifier).
+            value = MUST(value.as_object().get(identifier));
+
+            // 5. If value is undefined, return failure.
+            if (value.is_undefined())
+                return Error::from_string_literal("Value is undefined");
+        }
+    }
+
+    // 5. Assert: value is not an abrupt completion.
+    // NOTE: This is not possible in our implementation.
```

Comment:
This spec step seems wrong, as far as I can understand, since we are MUSTing our way through any steps that would return an abrupt completion

Reply:
Made a spec issue for this

Reply:
> This is not possible in our implementation.

I might have misunderstood this note and what you meant with this note is that *at this point there can be no abrupt completion (in our implementation)*.